pyincore_viz/geoutil.py

```diff
@@ -1094,6 +1094,78 @@ def get_ipyleaflet_heatmap(locations=None, radius=10, blur=10, max=1, name=""):
 
         return heatmap
 
+    @staticmethod
+    def plot_mulitple_vector_dataset(dataset_list, zoom_level=10):
+        geodata_dic_list = []
+        title_list = []
+        bbox = None
+        # check if the dataset is geodataset and convert dataset to geodataframe
+        for dataset in dataset_list:
+            try:
+                tmp_gpd = gpd.read_file(dataset.local_file_path)
+                tmp_min_x = tmp_gpd.bounds.minx.mean()
+                tmp_min_y = tmp_gpd.bounds.miny.mean()
+                tmp_max_x = tmp_gpd.bounds.maxx.mean()
+                tmp_max_y = tmp_gpd.bounds.maxy.mean()
+
+                if bbox is None:
+                    bbox = [tmp_min_x, tmp_min_y, tmp_max_x, tmp_max_y]
+                tmp_bbox = [tmp_min_x, tmp_min_y, tmp_max_x, tmp_max_y]
+
+                if bbox[0] >= tmp_bbox[0]:
+                    bbox[0] = tmp_bbox[0]
+                if bbox[1] >= tmp_bbox[1]:
+                    bbox[1] = tmp_bbox[1]
+                if bbox[2] <= tmp_bbox[2]:
+                    bbox[2] = tmp_bbox[2]
+                if bbox[3] <= tmp_bbox[3]:
+                    bbox[3] = tmp_bbox[3]
+
+                # skim geodataframe only for needed fields
+                tmp_fld_list = ['geometry']
+                tmp_gpd_skimmed = tmp_gpd[tmp_fld_list]
+                tmp_geo_data_dic = json.loads(tmp_gpd_skimmed.to_json())
+                geodata_dic_list.append(tmp_geo_data_dic)
+                title_list.append(dataset.metadata["title"])
+
+            except Exception:
+                raise("The dataset might not be a vector geodataset or field name might not be correct.")
```

Comment:
can you use Raise this way? Dont you need to add raise ValueError, etc?

Reply:
Raising value error

Reply:
could you do a search in the code base? I still see a few raise("") not related to this PR though